packages/calling/src/CallingClient/CallingClient.test.ts

```diff
@@ -234,254 +162,394 @@ describe('CallingClient Tests', () => {
+        file: 'CallingClient',
+        method: 'getMobiusServers',
+      });
+      expect(webex.request).toBeCalledOnceWith({
+        ...getMockRequestTemplate(),
+        uri: `${uri}?regionCode=${regionBody.clientRegion}&countryCode=${regionBody.countryCode}`,
+        method: 'GET',
+      });
+      expect(handleErrorSpy).not.toBeCalled();
+    });
+  });
+
+  describe('Network activity detection tests', () => {
+    let callingClient;
+    let line;
+    let reg;
+
+    beforeEach(async () => {
+      jest.useFakeTimers();
+
       webex.request
         .mockResolvedValueOnce(ipPayload)
         .mockResolvedValueOnce(regionPayload)
-        .mockRejectedValueOnce({statusCode: 404})
-        .mockResolvedValue(registrationPayload);
+        .mockResolvedValueOnce(discoveryPayload)
+        .mockResolvedValueOnce(registrationPayload);
+
+      callingClient = await createClient(webex, {logger: {level: LOGGER.INFO}});
+      line = Object.values(callingClient.lineDict)[0] as ILine;
+      reg = line.registration;
+
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.DEFAULT);
+      await line.register();
+    });
+
+    afterEach(() => {
+      jest.clearAllTimers();
+      jest.clearAllMocks();
+      callingClient.removeAllListeners();
+      callManager.removeAllListeners();
+      callingClient = undefined;
+      jest.useRealTimers();
+    });
+
+    it('detect a network flap in mercury connection', async () => {
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
 
-      expect(callingClient.getRegistrationStatus()).toBe(MobiusStatus.DEFAULT);
-      await callingClient.register();
-      await utils.waitForMsecs(50);
+      /* Set mercury connection to be down and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = false;
 
-      expect(handleErrorSpy).toHaveBeenCalled();
-      expect(callingClient.getRegistrationStatus()).toBe(MobiusStatus.ACTIVE);
+      jest.advanceTimersByTime(NETWORK_FLAP_TIMEOUT + 500);
 
-      const uri = `${webex.internal.services._serviceUrls.mobius}${URL_ENDPOINT}`;
-      expect(callingClient.getActiveMobiusUrl()).toEqual(uri);
+      /* We should be detecting the network flap */
       expect(warnSpy).toBeCalledWith(
-        'Error in finding Mobius Servers. Will use the default URL.',
-        ''
+        'Network has flapped, waiting for mercury connection to be up',
+        {file: CALLING_CLIENT_FILE, method: NETWORK_CHANGE_DETECTION_UTIL}
+      );
+
+      /* Set mercury connection to be up and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = true;
+
+      jest.advanceTimersByTime(NETWORK_FLAP_TIMEOUT + 500);
+
+      await flushPromises();
+
+      /* We should be detecting the network recovery */
+      expect(logSpy).nthCalledWith(
+        7,
+        'Mercury connection is up again, Re-registering with Mobius',
+        {file: REGISTRATION_FILE, method: 'handleConnectionRestoration'}
       );
```

Comment:
You may add these lines as well since at the beginning of test case: `line.getRegistrationStatus() was ACTIVE`

```suggestion
      );
       expect(deRegSpy).toBeCalledWith();
      expect(restoreSpy).toBeCalledWith('handleConnectionRestoration');
      expect(restartRegisterSpy).toBeCalledWith('handleConnectionRestoration');
      expect(webex.request).toBeCalledTimes(6);
      expect(registerSpy).toBeCalledWith('handleConnectionRestoration', [reg.getActiveMobiusUrl()]);
      expect(registerSpy).lastCalledWith('handleConnectionRestoration', [primaryUrl]);
```

Reply:
Added